x-pack/plugins/lens/public/editor_frame_service/embeddable/embeddable.tsx

```diff
@@ -233,4 +241,71 @@ export class Embeddable extends AbstractEmbeddable<LensEmbeddableInput, LensEmbe
       }
     }
   }
+
+  async initializeOutput() {
+    if (!this.savedVis) {
+      return;
+    }
+    const promises = this.savedVis.references
```

Comment:
I only moved this piece of code from `embeddable_factory` into `embeddable`, but it does make more sense to filter out the nulls here instead of after the `Promise.all`, so I've made that change